plutus-ledger-api/src/Plutus/V1/Ledger/EvaluationContext.hs

```diff
@@ -20,12 +29,24 @@ import Data.Maybe
 import Data.Set as Set
 import Data.Text qualified as Text
 import GHC.Exts (inline)
+import GHC.Generics
+
+type DefaultMachineParameters =
+    Plutus.MachineParameters CekMachineCosts CekValue DefaultUni DefaultFun
 
 -- | An opaque type that contains all the static parameters that the evaluator needs to evaluate a script.
 -- This is so that they can be computed once and cached, rather than recomputed on every evaluation.
-newtype EvaluationContext = EvaluationContext
-    { toMachineParameters :: Plutus.MachineParameters CekMachineCosts CekValue DefaultUni DefaultFun }
-    deriving newtype NFData
+data EvaluationContext = EvaluationContext
+    { machineParametersImmediate :: DefaultMachineParameters
+    , machineParametersDeferred  :: DefaultMachineParameters
+    }
+    deriving stock Generic
+    deriving anyclass NFData
+
+toMachineParameters :: EvaluationContext -> DefaultMachineParameters
+toMachineParameters = case defaultUnliftingMode of
+    UnliftingImmediate -> machineParametersImmediate
+    UnliftingDeferred  -> machineParametersDeferred
```

Comment:
in real-world (on-chain), do we have to keep both unlifting modes around? as long as we commit in a single one being the default, isn't that enough to keep around?

Reply:
I guess it does not matter that much since it is cached anyway.

Reply:
Yep, I'll do it.